### PR TITLE
[FINAL][HOLD] ZincContentBundleDeleteTask

### PR DESCRIPTION
**Pull request by msun**
moved to https://github.com/mindsnacks/Zinc-ObjC/pull/98

### CODE REVIEW

File: Zinc/Private/ZincContentBundleDeleteTask.m [hunk 1]
```diff
@@ -0,0 +1,294 @@
+//
```
Comment (msun) [inline Zinc/Private/ZincContentBundleDeleteTask.m:1]: this file is almost entirely copied from `NSFileManager+MSAdditions`

File: Zinc/Private/ZincContentBundleDeleteTask.m [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+- (BOOL)writeToRepoJSONWithJSONDict:(NSDictionary *)jsonDict {
+    [self deleteFileAtURL:self.repoJSONURL];
+
+    NSString *repoJSONPath = [self repoJSONPath];
+    NSError *error;
+
+    NSOutputStream *outputStream = [[NSOutputStream alloc] initToFileAtPath:repoJSONPath
+                                                                     append:NO];
+    [outputStream open];
+    [NSJSONSerialization writeJSONObject:jsonDict
```
Comment (aboedo) [inline Zinc/Private/ZincContentBundleDeleteTask.m:110]: you can also use zinc_writeToFile

File: Zinc/Private/ZincContentBundleDeleteTask.m [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+    NSURL *docDirURL = [self urlForSearchPathDirectory:NSDocumentDirectory];
+    NSURL *zincDirURL = [docDirURL URLByAppendingPathComponent:@"zinc"];
+    NSURL *url = [zincDirURL URLByAppendingPathComponent:path
+                                             isDirectory:isDirectory];
+    NSParameterAssert(url);
+    return url;
+
+}
+
+- (NSURL *)bundlesFolderURL {
```
Comment (aboedo) [inline Zinc/Private/ZincContentBundleDeleteTask.m:164]: any chance you can use ZincRepo's bundlesPath for this? same question with the others

File: Zinc/Private/ZincRepo.m [hunk 4]
```diff
@@ -134,8 +134,9 @@ + (instancetype) repoWithURL:(NSURL*)fileURL networkOperationQueue:(NSOperationQ
             if (![repo queueInitializationTasks]) {
                 repo.isInitialized = YES;
             }
-            
+
             [repo queueGarbageCollectTask];
+            [repo queueContentBundleDeleteTask];
```
Comment (aboedo) [inline Zinc/Private/ZincRepo.m:139]: 👍 
could you verify that this runs _before_ catalog update?

**Review by aboedo — COMMENTED** [review]
i think this approach should work well. good job

File: Zinc/Private/ZincRepo.m (hunk 4 above)
Comment (msun) [reply Zinc/Private/ZincRepo.m:139]: what is the reason for that?

Comment (aboedo) [reply Zinc/Private/ZincRepo.m:139]: actually, nevermind, there's no reason for that. forgot that this only deletes content.

Comment (msun) [reply Zinc/Private/ZincRepo.m:139]: :+1:

**Pull request closed without merging**